sassoftware/vscode-sas-extension
# feat: improve function autocomplete

**Summary**
- Do NOT show functions as snippets
  - They're built-in syntax and should be distinguished from other snippets
  - Snippets will show everywhere which is sometimes unexpected for functions
- Provide functions autocomplete similar to other keywords from syntax data
  - so that it will be more accurate and up-to-date (reported by #457)
- Provide function autocomplete when typed characters don't match other keyword but functions

_Note_: Functions can be used in various places mixed with other keywords, it can't be 100% match user's expectation. User can specifically turn this feature off by VS Code settings `Editor › Suggest: Show Functions`

**Testing**

- Make sure functions autocomplete don't show as snippets
- Make sure functions autocomplete don't appear in invalid places like string, comment or cards data
- Make sure functions autocomplete can appear when typing matched characters (like sleep/rannor reported in #457)
- Make sure corresponding help description can be shown with the function autocomplete (click `>` on the list item)
